src/main/java/hudson/maven/local_repo/PerExecutorLocalRepositoryLocator.java

```diff
@@ -15,7 +16,13 @@ public PerExecutorLocalRepositoryLocator() {}
 
     @Override
     public FilePath locate(AbstractMavenBuild build) {
-        return build.getBuiltOn().getRootPath().child("maven-repositories/"+ Executor.currentExecutor().getNumber());
+        final Node builtOn = build.getBuiltOn();
+        final FilePath rootPath = builtOn != null ? builtOn.getRootPath() : null;
+        final Executor executor = Executor.currentExecutor();
+        if (rootPath == null || executor == null) {
+            return null;
+        }
+        return rootPath.child("maven-repositories/"+ executor.getNumber());
```

Comment:
It looks like the above changes are just fixing an issue identified by Findbugs, right?